En el archivo auto.csv se encuentran los siguientes datos de diferentes automóviles: Cilindros, Cilindrada, Potencia, Peso, Aceleración, Año del coche, Origen, Consumo (mpg)

Las unidades de las características de los automóviles no se encuentran en el sistema internacional. La variable “origen” es un código que identifica al país de origen.

**Crea un modelo con él para que se pueda estimar el consumo de un vehículo a partir del resto de las variables.**

In [1]:
import pandas as pd
import numpy as np
from sklearn.linear_model import LinearRegression
import matplotlib.pyplot as plt

In [2]:
data_auto = pd.read_csv("auto.csv")
data_auto.head()

cylinders  displacement  horsepower  weight  acceleration  model_year  \
0          8         307.0       130.0  3504.0          12.0          70   
1          8         350.0       165.0  3693.0          11.5          70   
2          8         318.0       150.0  3436.0          11.0          70   
3          8         304.0       150.0  3433.0          12.0          70   
4          8         302.0       140.0  3449.0          10.5          70   

   origin   mpg  
0       1  18.0  
1       1  15.0  
2       1  18.0  
3       1  16.0  
4       1  17.0

In [3]:
data_auto.shape #392 filas y 8 columnas

(392, 8)

Voy a empezar haciendo un modelo de regresion lineal que ponga el mpg en funcion de todas las variables

In [4]:
X=data_auto[['cylinders','displacement','horsepower','weight','acceleration','model_year','origin']]
Y=data_auto['mpg']
lm = LinearRegression()
lm.fit(X, Y)

LinearRegression()

In [5]:
lm.score(X, Y) #asi tenemos R2

0.8214780764810599

In [6]:
lm.intercept_ #es alfa

-17.218434622017536

In [7]:
lm.coef_ #son los coeficientes de nuestro modelo (las betas)

array([-0.49337632,  0.01989564, -0.01695114, -0.00647404,  0.08057584,
        0.75077268,  1.4261405 ])

Así que el modelo que determina el consumo en función del resto de variables es:

**mpg=-17.21-0.49cylinders + 0.01displacement -0.01horsepower -0.006weight +0.08acceleration + 0.75model_year + 1.46origin**

con un R2 de 0.82, lo que hace que sea un modelo fiable